yapapi/net.py

```diff
@@ -0,0 +1,156 @@
+            "net": [
+                {
+                    "id": self.network.network_id,
+                    "ip": str(self.network.network_address),
+                    "mask": str(self.network.netmask),
+                    "nodeIp": str(self.ip),
+                    "nodes": self.network.nodes_dict,
+                }
+            ]
+        }
+        return deploy_args
+
+
+class Network:
+    @classmethod
+    async def create(
+        cls,
+        net_api: "Net",
+        ip: str,
+        owner_id: str,
+        owner_ip: Optional[str] = None,
+        network_id: Optional[str] = None,
+        mask: Optional[str] = None,
+        gateway: Optional[str] = None,
+    ):
+        network = cls(net_api, ip, owner_id, owner_ip, network_id, mask, gateway)
+        # create the network in yagna
+        await net_api.create_network(network)
+        # add requestor's own address to the network
+        await network.add_address(network.owner_ip)
+        return network
+
+    def __init__(
+        self,
+        net_api: "Net",
+        ip: str,
+        owner_id: str,
+        owner_ip: Optional[str] = None,
+        network_id: Optional[str] = None,
+        mask: Optional[str] = None,
+        gateway: Optional[str] = None,
+    ):
+        self._net_api = net_api
+        network_ip = f"{ip}/{mask}" if mask else ip
+        try:
+            self._ip_network: IpNetwork = ip_network(network_ip, strict=False)
+        except ValueError as e:
+            raise NetworkError(f"{e}.")
+        self._hosts = self._ip_network.hosts()
+
+        self._network_id = network_id or "0x" + uuid4().hex
+        self._gateway = gateway
+        self._owner_id = owner_id
+        self._owner_ip = owner_ip or self._next_address()
+        self._nodes: Dict[str, Node] = dict()
+
+    def __str__(self) -> str:
+        return f"""Network {{
+        id: {self._network_id}
+        ip: {self.network_address}
+        mask: {self.netmask}
+        owner_ip: {self._owner_ip}
+        nodes: {self.nodes_dict}
+    }}"""
+
+    @property
+    def owner_ip(self) -> str:
+        return str(self._owner_ip)
+
+    @property
+    def network_address(self) -> str:
+        return str(self._ip_network.network_address)
+
+    @property
+    def netmask(self) -> str:
+        return str(self._ip_network.netmask)
+
+    @property
+    def gateway(self) -> Optional[str]:
+        if self._gateway:
+            return str(self._gateway)
+        return None
+
+    @property
+    def nodes_dict(self) -> Dict[str, str]:
+        return {str(v.ip): k for k, v in self._nodes.items()}
+
+    @property
+    def network_id(self):
+        return self._network_id
+
+    def _ensure_ip_in_network(self, ip: str):
+        if ip_address(ip) not in self._ip_network:
+            raise NetworkError(
+                f"The given IP ('{ip}') address must belong to the network ('{self._ip_network.with_netmask}')."
+            )
+
+    async def add_address(self, ip: str):
```

Comment:
I would rename this to `add_owner_address`, or `add_requestor_address`.

Reply:
done